### PR TITLE
added a progress hook

### PR DESCRIPTION
**Pull request by prairir**
### What are you trying to accomplish?
simplify the way we use progress

### How are you accomplishing it?
created a hook to extract and simplify the changing and using of the progress

### Is there anything reviewers should know?
it isnt tested yet but i will comment when it is tested

- [x] Is it safe to rollback this change if anything goes wrong?

### CODE REVIEW

File: app/javascript/scavenger_hunt/hooks/useProgress/useProgress.jsx [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+      })
+    }
+  }
+
+  const currentProgress = (isLoggedIn) ? parseInt(userData.currentUser.progress) : 0
+
+  return [ currentProgress, setProgress, isLoading ];
+}
+
+export { useProgress }
```
Comment (amanpatel123) [inline app/javascript/scavenger_hunt/hooks/useProgress/useProgress.jsx:25]: Beautiful !🙌🏻

File: app/javascript/scavenger_hunt/scenes/HomePage/HomePage.jsx [hunk 2]
```diff
@@ -1,16 +1,22 @@
...
-import './HomePage.scss'
+import React from 'react';
 import { Button } from 'react-bootstrap';
 
+import { useProgress } from '../../hooks/useProgress'
+import { Alert } from "../../components/Alert";
+import './HomePage.scss';
+
 const HomePage = (props) => {
+  const [progress, _, isProgressLoading] = useProgress((props.location.state && props.location.state.loggedIn));
```
Comment (amanpatel123) [inline app/javascript/scavenger_hunt/scenes/HomePage/HomePage.jsx:9]:
```suggestion
  const [progress, _ , isProgressLoading] = useProgress((props.location.state && props.location.state.loggedIn));
```

File: app/javascript/scavenger_hunt/scenes/PlayArena/PlayArena.jsx [hunk 3]
```diff
@@ -21,12 +20,6 @@ const PlayArena = (props) => {
     return clueId
   }
 
-  if (queryLoading) {
-    return(
-      <h3>Loading your data</h3>
-    )
-  }
-
   if (!queryLoading && !userData.currentUser) {
```
Comment (amanpatel123) [inline app/javascript/scavenger_hunt/scenes/PlayArena/PlayArena.jsx:23]: Do we have `queryLoading` variable now? should it not be `isLoading`? Which we get from `useProgress` Hook

File: app/javascript/scavenger_hunt/scenes/PlayArena/PlayArena.jsx [hunk 4]
```diff
@@ -36,10 +29,17 @@ const PlayArena = (props) => {
    )
   }
 
+  const [progress, setProgress, isProgressLoading] = useProgress(true);
+
+  if (queryLoading || isProgressLoading) {
```
Comment (amanpatel123) [inline app/javascript/scavenger_hunt/scenes/PlayArena/PlayArena.jsx:34]: Same comment here `queryLoading` is not defined

File: app/javascript/scavenger_hunt/scenes/PlayArena/PlayArena.jsx (hunk 3 above)
Comment (prairir) [reply app/javascript/scavenger_hunt/scenes/PlayArena/PlayArena.jsx:23]: whoops, i didnt see that. thanks il change that.

**Comment by prairir** [comment]
this is bad and we are going another way

**Pull request closed without merging**